cmd/kubelet/app/server.go

```diff
@@ -1295,18 +1295,37 @@ func newTracerProvider(s *options.KubeletServer) (oteltrace.TracerProvider, erro
 
 func getCgroupDriverFromCRI(ctx context.Context, s *options.KubeletServer, kubeDeps *kubelet.Dependencies) error {
 	klog.V(4).InfoS("Getting CRI runtime configuration information")
-	runtimeConfig, err := kubeDeps.RemoteRuntimeService.RuntimeConfig(ctx)
-	if err != nil {
-		s, ok := status.FromError(err)
-		if !ok || s.Code() != codes.Unimplemented {
-			return err
+
+	var (
+		runtimeConfig *runtimeapi.RuntimeConfigResponse
+		err           error
+	)
+	// Retry a couple of times, hoping that any errors are transient.
+	// Fail quickly on known, non transient errors.
+	for i := 0; i < 3; i++ {
```

Comment:
I am not sure that adding a loop here is adding much value. What transient errors do we expect to succeed on retries? Also, there is no backoff or wait in between attempts.

Reply:
if kubelet and runtime are started together, transient issue may be just a timing of a start. Before we were establishing connection a bit later in a function - after reading some metrics and such. We can exit kubelet immediately and rely on external restart of `kubelet`. Or we can do a few iterations here. But if we do retry here - there should be a delay between attempts

Reply:
Yes, without delay there isn't much value here.

Reply:
how is 2 seconds? I had trouble finding a value to base it on so I just made a guess.

Reply:
I'm back from PTO, thanks for all the help with this @kubernetes/milestone-maintainers 

How long would we expect for the CRI runtime startup to take? Just wonder should the sleep be 1 or 2 seconds or smth else.